src/test/util/event_buffer.rs

```diff
@@ -305,12 +275,25 @@ impl<'a, T: Clone> EventSubscriber<'a, T> {
         None
     }
 
-    /// Consume and pass events to the provided closure until it returns Some or the timeout is hit.
-    pub(crate) async fn filter_map_event<F, R>(
-        &mut self,
-        timeout: Duration,
-        mut filter_map: F,
-    ) -> Option<R>
+    /// Get the next unread event from the underlying buffer, waiting for a new one to arrive if all
+    /// current ones have been read.
+    pub(crate) async fn next(&mut self, timeout: Duration) -> Option<T> {
+        crate::runtime::timeout(timeout, async move {
+            loop {
+                let notified = self.buffer.inner.event_received.notified();
+                if let Some(next) = self.try_next() {
+                    return Some(next);
+                }
+                notified.await;
+            }
+        })
+        .await
+        .unwrap_or(None)
+    }
+
+    /// Get the next unread event for which the provided closure returnes `Some`, waiting for new
+    /// events to arrive if all current ones have been read.
+    pub(crate) async fn next_map<F, R>(&mut self, timeout: Duration, mut filter_map: F) -> Option<R>
```

Comment:
Along with renaming the struct, I renamed a bunch of its method to better match the iteration/streaming model.